src/common/transformations/src/transformations/op_conversions/eye_decomposition.cpp

```diff
@@ -0,0 +1,123 @@
+};
+
+std::shared_ptr<Node> EyeDecomposition::make_eye_model(const Output<Node>& height,
+                                                       const Output<Node>& width,
+                                                       const Output<Node>& k,
+                                                       element::Type dtype) {
+    const auto zero_int = register_new_node(opset9::Constant::create(element::i64, Shape{1}, {0}));
+    const auto zero = register_new_node(opset9::Constant::create(dtype, Shape{1}, {0}));
+    const auto one = register_new_node(opset9::Constant::create(dtype, Shape{1}, {1}));
+
+    const auto k_neg = register_new_node<opset9::Negative>(k);
+    const auto k_axis = register_new_node<opset9::Concat>(OutputVector{k_neg, k}, 0);
+
+    const auto eye_shape = register_new_node<opset9::Concat>(OutputVector{height, width}, 0);
+
+    // Calculate eye zero padding and internal square eye size.
+    const auto pad_start =
+        register_new_node<opset9::Minimum>(eye_shape, register_new_node<opset9::Maximum>(zero_int, k_axis));
+    const auto shape_pad_diff = register_new_node<opset9::Subtract>(eye_shape, pad_start);
+    const auto eye_size = register_new_node<opset9::ReduceMin>(shape_pad_diff, zero_int, true);
+    const auto pad_end = register_new_node<opset9::Subtract>(shape_pad_diff, eye_size);
+
+    // Make 1d-eye as eye_size times of (1, zeros(eye_size)), trimmed at end by eye_size elements.
+    const auto zeros = register_new_node<opset9::Tile>(zero, eye_size);
+    const auto one_followed_by_zeros = register_new_node<opset9::Concat>(OutputVector{one, zeros}, 0);
+    const auto eye_1d = register_new_node<opset9::Pad>(register_new_node<opset9::Tile>(one_followed_by_zeros, eye_size),
+                                                       zero_int,
+                                                       register_new_node<opset9::Negative>(eye_size),
+                                                       op::PadMode::CONSTANT);
+    // Reshape 1d-eye to 2d-eye
+    const auto eye_2d =
+        register_new_node<opset9::Reshape>(eye_1d,
+                                           register_new_node<opset9::Concat>(OutputVector{eye_size, eye_size}, 0),
+                                           false);
+
+    // Pad Eye to get final shape
+    return register_new_node<opset9::Pad>(eye_2d, pad_start, pad_end, op::PadMode::CONSTANT);
+}
+
+std::shared_ptr<Node> EyeDecomposition::make_eye_batches(const Output<Node>& eye, const Output<Node>& batch) {
+    const auto eye_tile = register_new_node<opset9::Constant>(element::i64, Shape{2}, 1);
+
+    // `batch_repeats` repeat eye matrix as tile only in higher dimensions than 1 by number(s) in batch parameter.
+    const auto batch_repeats = register_new_node<opset9::Concat>(OutputVector{batch, eye_tile}, 0);
+
+    return register_new_node<opset9::Tile>(eye, batch_repeats);
+}
+
+EyeDecomposition::EyeDecomposition() {
```

Comment:
please add this macro here
MATCHER_SCOPE(EyeDecomposition);

Reply:
Added.